### PR TITLE
Benchmark cmake: Allow building from a Swift root.

### PR DESCRIPTION
**Pull request by atrick**
And fix the SWIFT_LIBRARY_PATH option so that it also sets the
RPATH.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci benchmark.

**Comment by atrick** [comment]
@swift-ci benchmark

**Comment by atrick** [comment]
@swift-ci benchmark

**Comment by atrick** [comment]
@swift-ci test.

**Comment by atrick** [comment]
@gottesmm I'm done with these cmake fixes. I didn't do any auxiliary cleanup. It should be trivial to review.

**Review by gottesmm — COMMENTED** [review]
Some quick questions. Then I want to look one more time.

**Comment by atrick** [comment]
@gottesmm Getting all the right behavior without adding a new option was too tricky. So now we have 
SWIFT_BENCHMARK_USE_OS_LIBRARIES, and no weird inference during configuration.

There's nothing we can really infer from the SWIFT_BENCHMARK_BUILT_STANDALONE flag, except that SWIFT_LIBRARY_PATH was not already set by the LLVM config.

**Comment by atrick** [comment]
@swift-ci benchmark

File: benchmark/CMakeLists.txt [hunk 1]
```diff
@@ -268,10 +271,18 @@ configure_sdks()
 message("--")
 message("-- Swift Benchmark Suite:")
 message("--   SWIFT_BENCHMARK_BUILT_STANDALONE = ${SWIFT_BENCHMARK_BUILT_STANDALONE}")
-message("--   SWIFT_EXEC = ${SWIFT_EXEC}")
-message("--   SWIFT_BENCHMARK_EXTRA_FLAGS = ${SWIFT_BENCHMARK_EXTRA_FLAGS}")
+message("--   SWIFT_BENCHMARK_USE_OS_LIBRARIES = ${SWIFT_BENCHMARK_USE_OS_LIBRARIES}")
+message("--   SWIFT_EXEC         = ${SWIFT_EXEC}")
 message("--   SWIFT_LIBRARY_PATH = ${SWIFT_LIBRARY_PATH}")
-message("--   CLANG_EXEC = ${CLANG_EXEC}")
+if (SWIFT_RPATH_BASE)
```
Comment (atrick) [inline benchmark/CMakeLists.txt]: Add a top-level comment explaining what an rpath base is

**Comment by atrick** [comment]
@swift-ci benchmark

**Comment by atrick** [comment]
@swift-ci test

**Comment by atrick** [comment]
@swift-ci benchmark

**Comment by atrick** [comment]
@swift-ci benchmark benchmark benchmark

**Comment by atrick** [comment]
@swift-ci test

**Comment by atrick** [comment]
@swift-ci benchmark

**Comment by atrick** [comment]
@swift-ci test OSX platform

**Comment by atrick** [comment]
@swift-ci test OS X

**Pull request merged**